frontend_tests/node_tests/message_flags.js

```diff
@@ -169,6 +169,16 @@ run_test("read", (override) => {
         },
         success: channel_post_opts.success,
     });
+
+    // verify early return on empty data
```

Comment:
Can we put this in its own `run_test` wrapper?  (even if it requires a bit extra setup, I like the early-return stuff to be easy to identify)